src/main/java/io/github/thebusybiscuit/slimefun4/utils/NumberUtils.java

```diff
@@ -281,4 +281,34 @@ public static int getJavaVersion() {
             return 0;
         }
     }
+    
+    /**
+     * This detects if 2 integers will overflow/underflow and if they will, returns the corresponding value
+     * @param a the first integer
+     * @param b the second integer
+     * @return {@link Integer#MAX_VALUE} if overflow detected, {@link Integer#MIN_VALUE} if underflow detected, otherwise the sum of a and b
+     */
+    public static int flowSafeAddition(int a, int b) {
+        return limitedAddition(a, b, Integer.MAX_VALUE, Integer.MIN_VALUE);
+    }
+    
+    /**
+     * This detects if 2 integers will overflow/underflow past a maximum or minimum value and if they will, returns the corresponding value
+     * @param a the first integer
+     * @param b the second integer
+     * @param max the maximum value for the operation
+     * @param min the minimum value for the operation
+     * @return max if overflow detected, min if underflow detected, otherwise the sum of a and b
+     */
+    public static int limitedAddition(int a, int b, int max, int min) {
```

Comment:
Some voice inside my head wants `min` to come before `max`.

```suggestion
     * @param min the minimum value for the operation
     * @param max the maximum value for the operation
     * @return max if overflow detected, min if underflow detected, otherwise the sum of a and b
     */
    public static int limitedAddition(int a, int b, int min, int max) {
```

Reply:
Now that you say that I 100% agree